thunder/tests/test_elementwise.py

```diff
@@ -61,36 +78,58 @@ def filter_b_values(vals, special):
             return shift_inps
         return vals
 
-    for op, a_types, b_types, special in elementwise_binary_ops:
+    def foo(a, b):
+        return op(a, b)
 
-        def foo(a, b):
-            return op(a, b)
+    cfoo = executor.make_callable(foo, cache=cache_option)
 
-        cfoo = executor.make_callable(foo)
+    a_vals = gather_inputs(a_types)
+    b_vals = filter_b_values(gather_inputs(b_types), special)
 
-        a_vals = gather_inputs(a_types)
-        b_vals = filter_b_values(gather_inputs(b_types), special)
+    for a in a_vals:
+        for b in b_vals:
+            actual = cfoo(a, b)
+            expected = foo(a, b)
+            assert_close(actual, expected)
+
+    if cache_option == "symbolic values":
+        assert thunder.cache_misses(cfoo) == len(a_types) * len(b_types)
 
-        for a in a_vals:
-            for b in b_vals:
-                actual = cfoo(a, b)
-                expected = foo(a, b)
-                assert_close(actual, expected)
+        fixed_a = random.choice(a_vals)
```

Comment:
What are we trying to test here? I think product(fixed_a, b_vals) is already tested above.